test/framework/autoscaler_helpers.go

```diff
@@ -249,7 +287,7 @@ func ProcessYAML(input *ProcessYAMLInput) ([]byte, error) {
 	return out, nil
 }
 
-type DisableAutoscalerForMachineDeploymentTopologyAndWaitInput struct {
+type DisableAutoscalerForMachineTopologyAndWaitInput struct {
```

Comment:
Is it necessary to share this struct or are we just saving copying 5 lines? If it's the latter please duplicate the struct (I don't really like the name, machine topology is not a thing, but I know we did something similar to that in some other CC+MP PRs, still would like to not make this a pattern, especially in a public API)

Same for EnableAutoscalerForMachineTopologyAndWaitInput